### PR TITLE
Ap 330/feature v3

### PR DESCRIPTION
**Pull request by ccleung**
WIP

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -11,7 +11,8 @@ Description: Functions for interacting directly with the Quandl API to offer dat
     in a number of formats usable in R, as well as the ability to search.
 Imports:
     httr (>= 0.6.1),
-    zoo
+    zoo,
+    jsonlite
 Suggests:
```
Comment (dmpe) [inline DESCRIPTION]: xts, timeSeries should also go into the `@imports`. Like here https://github.com/quandl/quandl-r/pull/36/files

File: R/Quandlapi.R [hunk 2]
```diff
@@ -1,63 +1,82 @@
...
 #' @export
-quandl.api <- function(version="v1", path, http = c('GET', 'PUT', 'POST', 'DELETE'), postdata = NULL, ...) {
+quandl.api <- function(path, http = c('GET', 'PUT', 'POST', 'DELETE'), postdata = NULL, ...) {
+  http <- match.arg(http)
+  request <- quandl.api.build_request(path, ...)
+  response <- httr::VERB(http, request$request_url, config = do.call(httr::add_headers, request$headers),
+                                      body = postdata, query = request$params)
+
+  quandl.api.handl_errors(response)
+  text_response <- httr::content(response, as="text")
```
Comment (dmpe) [inline R/Quandlapi.R]: Please, could you also use everywhere where it is needed: 
```
#' @import httr
#' @import jsonlite
```
E.g. like here https://github.com/quandl/quandl-r/pull/36/files#diff-185842ab024dc31efeaf6b40e8455b34R56

**Comment by RaymondMcT** [comment]
xts needs to stay in depends as the package will break if it is moved into imports. If I move timeSeries into imports it will be automatically installed with Quandl. It is the intention to expand the return formats eventually to some of the others in R (its, irts, etc.) and I do not want them to be required installations with the package.

As well adding the lines

    #' @import httr
    #' @import jsonlite

Will load the entirety of httr and jsonlite into the global namespace, which I don't think is necessary.

**Comment by dmpe** [comment]
In regard to the second one: you can also use importFrom pck fnct. 



-> timeseries: then it should be documented (+at least added to the 
suggested ones)

**Comment by ccleung** [comment]
@dmpe Is there a reason why you want to import the namespace directives? we are hoping to following best practices seen in other popular R packages, e.g., `devtools` does not import package namespaces: https://github.com/hadley/devtools/blob/master/R/github.R#L10 and we prefer not to do so as well

**Comment by dmpe** [comment]
@ccleung 

1.
Yes.
https://github.com/ropensci/onboarding/issues/17#issuecomment-116782430 (section Imports)

`
You have httr and jsonlite (and many others) as Imports in DESCRIPTION file, but they aren't listed in any @ import. Was there a reason behind this? Even though httr and jsonlite are namespaced, those will still fail for users that don't have those pkgs installed.
`

2.
`best practices seen in other popular R packages:` Well, dplyr is very popular. 

https://github.com/hadley/dplyr/search?utf8=%E2%9C%93&q=importFrom&type=Code
https://github.com/ropensci/geojsonio/search?utf8=%E2%9C%93&q=import&type=Code
https://github.com/ropensci/fulltext/search?utf8=%E2%9C%93&q=import

**Pull request merged**